bridges/otelslog/handler.go

```diff
@@ -0,0 +1,93 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+// Package otelslog provides [Handler], an [slog.Handler] implementation, that
+// can be used to bridge between the [log/slog] API and [OpenTelemetry].
+//
+// [OpenTelemetry]: https://opentelemetry.io/docs/concepts/signals/logs/
+package otelslog // import "go.opentelemetry.io/contrib/bridges/otelslog"
+
+import (
+	"context"
+	"log/slog"
+
+	"go.opentelemetry.io/otel/log"
+	"go.opentelemetry.io/otel/sdk/instrumentation"
+)
+
+type config struct{}
+
+// Option configures a [Handler].
+type Option interface {
+	apply(config) config
+}
+
+type optFunc func(config) config
+
+func (f optFunc) apply(c config) config { return f(c) }
+
+// WithInstrumentationScope returns an option that configures the scope of the
+// [log.Logger] used by a [Handler].
+//
+// By default if this Option is not provided, the Handler will use a default
+// instrumentation scope describing this bridge package. It is recommended to
+// provide this so log data can be associated with its source package or
+// module.
+func WithInstrumentationScope(scope instrumentation.Scope) Option {
+	return optFunc(func(c config) config {
+		// TODO: implement.
+		return c
+	})
+}
+
+// Handler is an [slog.Handler] that sends all logging records it receives to
+// OpenTelemetry.
+type Handler struct {
+	// Ensure forward compatibility by explicitly making this not comparable.
+	noCmp [0]func() //nolint: unused  // This is indeed used.
+
+	// TODO: implement.
+}
+
+// Compile-time check *Handler implements slog.Handler.
+var _ slog.Handler = (*Handler)(nil)
+
+// New returns a new [Handler] to be used as an [slog.Handler].
+//
+// If lp is nil, [noop.LoggerProvider] will be used as the default.
+//
+// By default the returned Handler will use a [log.Logger] that is identified
+// with this bridge package information. [WithInstrumentationScope] should be
+// used to override this with details about the package or module the handler
+// will instrument.
+func New(lp log.LoggerProvider, options ...Option) *Handler {
```

Comment:
Should this be `NewHandler`?  The package doesn't include handler in the name.  `otelslog.NewHandler()` reads better to me than `otelslog.New()`, since slog contains other interfaces.

Reply:
Sounds good to me. I think we can have both `New` and `NewHandler`. The `New` function can return an `slog.Logger` instance. Thoughts @dashpole?

Reply:
I'm OK with that.  WDYT about NewLogger vs New?  Standard usage would be:

* `slog.SetDefault(otelslog.New())` vs
* `slog.SetDefault(otelslog.NewLogger())`

Reply:
Update. PTAL @dashpole